x/blob/types/blob_tx.go

```diff
@@ -15,7 +15,7 @@ type Blob = tmproto.Blob
 
 // NewBlob creates a new coretypes.Blob from the provided data after performing
 // basic stateless checks over it.
-func NewBlob(ns appns.Namespace, blob []byte) (*Blob, error) {
+func NewBlob(ns appns.Namespace, blob []byte, share uint8) (*Blob, error) {
```

Comment:
[nit]
```suggestion
func NewBlob(ns appns.Namespace, blob []byte, shareVersion uint8) (*Blob, error) {
```

[blocking] since this function accepts a new parameter, I would accept it to use the `shareVersion` below:

```diff
	return &tmproto.Blob{
		NamespaceId:      ns.ID,
		Data:             blob,
-		ShareVersion:     uint32(appconsts.DefaultShareVersion),
+               ShareVersion:     uint32(shareVersion),
		NamespaceVersion: uint32(ns.Version),
	}, nil
```

Additionally this function should verify that the shareVersion argument is present in https://github.com/celestiaorg/celestia-app/blob/ab24a7b4d5190a53cf4bb0a9e989f4c824759f8e/pkg/appconsts/appconsts.go#L103 prior to returning the Blob

Reply:
got it.